lib/transactions/forms.py

```diff
@@ -51,7 +62,12 @@ def __init__(self, *args, **kw):
         super(UpdateForm, self).__init__(*args, **kw)
 
     def clean(self):
-        keys = set(self.data.keys()).difference(set(self.fields.keys()))
+        fields = self.fields.keys()
+        # If the provider is already set, don't allow it to be changed.
+        if self.old.get('provider'):
```

Comment:
It doesn't delete the data, if deletes the field off the form. So if you set provider and then try and patch provider it throws a ValidationError on the next lines (https://github.com/mozilla/solitude/pull/322/files#diff-ecc0a1a041d0a2d3c4e9555c196ab6b8R73). The deleting of the field off the form is an internal thing, but this is a quick way to ensure that from a calling API point of view, you'll get a 400 on the second attempt. I've added a test to demonstrate this.